core/modules/wildcard_match.cc

```diff
@@ -335,7 +335,10 @@ int WildcardMatch::DelEntry(struct WmTuple *tuple, wm_hkey_t *key) {
     tuple->ht.Close();
 
     num_tuples_--;
-    memmove(&tuples_[idx], &tuples_[idx + 1],
+
+    // shift remaing tuples to the left
+    memmove(reinterpret_cast<void*>(&tuples_[idx]),
```

Comment:
Perhaps we can devise a common framework for the performance of every module?
* We define a few input sets (packet batches), each of which represents a combination of various packets
* Benchmark is done with every combination of input set and module (thus producing a table)
  * If the kNumIGates of a module is zero we can skip the module.
  * We create a dummy module, and connect the module under test to the dummy (via each of kNumOGates output gates).
* The benchmark code directly call module's ProcessBatch() directly.
* It can be done with google benchmark or our custom code. The latter case is interesting in that:
  * We can measure the latency at the cycle granularity with `rdtscp`. This is beneficial since the wall-clock time used by gbenchmark is not constant)
  * All other overheads (google benchmark, packet alloc/free, etc.) can be excluded.
  * min/median latency for each call can be calculated, which will be much more consistent across runs than average, without noise.